File: src/main/java/com/example/model/Instructor.java
package com.example.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class Instructor {
    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
    private int instructorId;

    private String name;
    private String email;
    private String contact;
    private String address;

    @ManyToOne
    @JoinColumn(name = "userId")
    private User user;



	public int getInstructorId() {
		return instructorId;
	}

	public void setInstructorId(int instructorId) {
		this.instructorId = instructorId;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getContact() {
		return contact;
	}

	public void setContact(String contact) {
		this.contact = contact;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	@Override
	public String toString() {
		return "Instructor [instructorId=" + instructorId + ", name=" + name + ", email=" + email + ", contact="
				+ contact + ", address=" + address + ", user=" + user + "]";
	}
    
    
}

File: Frontend/src/app/model/Instructor.ts
// Instructor.ts
import { User } from "./user";
import { Zym } from "./Zym";
import { WorkoutPlan } from "./WorkoutPlan"; 
export class Instructor {
  instructorId: number ;
  name: string;
  email: string;
  contact: string;
  address: string;
  user: User;
  zym: Zym;
  workoutPlans : WorkoutPlan[]; 
  constructor() {
    this.instructorId = null as any;
    this.name = '';
    this.email = '';
    this.contact = '';
    this.address = '';
    this.user = new User();
    this.zym = new Zym();
    this.workoutPlans = [];
  }
}